src/contracts/proxies/Vault721.sol

```diff
@@ -262,7 +265,7 @@ contract Vault721 is ERC721EnumerableUpgradeable {
   /**
    * @dev check that proxy does not exist OR that the user does not own proxy
    */
-  function _isNotProxy(address _user) internal view returns (bool) {
+  function _buildProxy(address _user) internal view returns (bool) {
```

Comment:
I still don't like the naming here. `_buildProxy` sounds like an action, but we are asking a question. Do you like "_shouldBuild" or "_shouldBuildProxy"?